### PR TITLE
User story35

### PR DESCRIPTION
**Pull request by andreaamato-polito**

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -54,12 +60,68 @@ app.use(passport.initialize());
...
+            name: name,
+            time: time
+        });
+        res.end();
+    } catch (error) {
+        res.status(500).json(error);
+    }
+});
+
+app.put('/api/services/:id', /* isAdmin, */ async (req, res) => {
```
Comment (fede1997483) [inline server/server.js:98]: Not useful for the story, i suggest to remove it before merging the code.

File: server/server.js [hunk 2]
```diff
@@ -54,12 +60,68 @@ app.use(passport.initialize());
...
+            name: name,
+            time: time
+        });
+        res.end();
+    } catch (error) {
+        res.status(500).json(error);
+    }
+});
+
+app.delete('/api/services/:id', /* isAdmin, */ async (req, res) => {
```
Comment (fede1997483) [inline server/server.js:115]: Can be removed.

File: server/dao.js [hunk 3]
```diff
@@ -22,4 +22,112 @@ exports.getTeam = () => {
...
+            if (err) {
+                reject(err);
+                return;
+            }
+            resolve(this.lastID);
+        });
+    });
+}
+
+exports.modifyService = (service) => {
```
Comment (fede1997483) [inline server/dao.js:80]: As in the server.js file, i suggest to remove this before merging.

File: server/dao.js [hunk 4]
```diff
@@ -22,4 +22,112 @@ exports.getTeam = () => {
...
+                reject(err);
+                return;
+            }
+            resolve(this.lastID);
+        });
+    });
+};
+
+
+exports.deleteService = (serviceId) => {
```
Comment (fede1997483) [inline server/dao.js:96]: Can be removed.

File: client/src/API.js [hunk 5]
```diff
@@ -43,7 +43,20 @@ async function getUserInfo() {
     }
 }
 
+async function createService(service) {
```
Comment (fede1997483) [inline client/src/API.js:46]: This looks good. Creating a service it's the goal of user story 35.

File: client/src/App.js [hunk 6]
```diff
@@ -127,14 +110,75 @@ function LoginForm(props) {
...
+      console.log('Invalid form');
+    }
+  }
+
+  return (
+    <Form>
+      <Row>
+
+        <Col className='alignCenter'>
+          <Form.Group className="mb-3" controlId="formServiceName">
```
Comment (fede1997483) [inline client/src/App.js:155]: Maybe in this case using a dropbox menu it's a little better for user experience.

File: client/src/App.js [hunk 7]
```diff
@@ -127,14 +110,75 @@ function LoginForm(props) {
...
+            <Form.Control
+              type="text"
+              placeholder="Enter service name"
+              onChange={ev => setName(ev.target.value)}
+            />
```
Comment (fede1997483) [inline client/src/App.js:160]: Maybe in this case using a dropbox menu it's a little better for user experience.

File: server/dao.js [hunk 8]
```diff
@@ -22,4 +22,112 @@ exports.getTeam = () => {
...
+        const sql = 'DELETE FROM services WHERE id=?';
+        db.run(sql, [serviceId], (err) => {
+            if (err) {
+                reject(err);
+                return;
+            }
+            resolve(this.lastID);
+        });
+    });
+}
```
Comment (fede1997483) [inline server/dao.js:107]: As in the server.js file, i suggest to remove this before merging.

File: server/server.js [hunk 9]
```diff
@@ -54,12 +60,68 @@ app.use(passport.initialize());
...
+    const id = req.params.id;
+    try {
+        await dao.deleteService(id);
+        res.end();
+    } catch (error) {
+        res.status(500).json(error);
+    }
+});
+
```
Comment (fede1997483) [inline server/server.js:124]: Not useful for the story, i suggest to remove it before merging the code.

Comment (andreaamato-polito) [reply server/server.js:124]: removed.

**Review by fede1997483 — APPROVED** [review]

**Pull request merged**